catalyst_voices/lib/widgets/app_bar/session/session_state_header.dart

```diff
@@ -21,15 +24,31 @@ class SessionStateHeader extends StatelessWidget {
           GuestSessionState() => const _GuestButton(),
           ActiveUserSessionState(:final user) => AccountPopup(
               avatarLetter: user.acronym,
-              onLockAccountTap: () => debugPrint('Lock account'),
-              onProfileKeychainTap: () => unawaited(
-                const AccountRoute().push<void>(context),
-              ),
+              onLockAccountTap: () => _onLockAccount(context),
+              onProfileKeychainTap: () => _onSeeProfile(context),
             ),
         };
       },
     );
   }
+
+  void _onLockAccount(BuildContext context) {
+    context.read<SessionBloc>().add(const LockSessionEvent());
+
+    VoicesSnackBar(
+      type: VoicesSnackBarType.error,
+      behavior: SnackBarBehavior.floating,
+      icon: VoicesAssets.icons.lockClosed.buildIcon(),
+      title: context.l10n.lockSnackbarTitle,
+      message: context.l10n.lockSnackbarMessage,
+    ).show(context);
```

Comment:
while at the moment its okey we should have TODO somewhere and comeback to it. 

Correct way to do it is to have somewhere listener on `SessionBloc`, like in `UnlockKeychainDialog`, but maybe global one, which will show those snackbars.

Reply:
I'm fixing it now, actually that's good idea. I will put it at app level

Reply:
I had to put the GlobalSessionListener under `MaterialApp builder`, cannot wrap the whole `MaterialApp` because only a child of `MaterialApp` can use localizations: `context.l10n`

```
  builder: (context, child) {
        return GlobalPrecacheImages(
          child: GlobalSessionListener(
            child: child ?? const SizedBox.shrink(),
          ),
        );
      },
```